tests/query_planner/test_query_planner.py

```diff
@@ -324,3 +324,39 @@ async def testable_generate_plans():
         testable_generate_plans,
         "generate plan for a typical query graph (50k KPs)",
     )
+
+
+@pytest.mark.asyncio
+@with_registry_overlay(settings.kpregistry_url, kps_from_string(
+    """
+    kp0 biolink:Disease -biolink:treated_by-> biolink:Drug
+    """
+))
+@with_norm_overlay(settings.normalizer_url)
+async def test_bad_norm():
+    """
+    Test that the pinned node "XXX:123" that the normalizer does not know
+    is still handled correctly based on the provided category.
+    """
+
+    qg = {
+        "nodes": {
+            "n0": {
+                "id": "XXX:123",
```

Comment:
Could you use the new query graph string utility for consistency?

Reply:
I don't think so... The query graph string function appears to take only a single property per node, we need two here. We could probably improve it.

Reply:
Ah good point. I didn't notice that.